src/rvsdg/from_cfg.rs

```diff
@@ -421,11 +421,20 @@ impl<'a> RvsdgBuilder<'a> {
                     pos,
                     op_type,
                 } => match op {
-                    ValueOps::Alloc | ValueOps::Load | ValueOps::PtrAdd => {
-                        return Err(RvsdgError::UnsupportedOperation {
-                            op: *op,
-                            pos: pos.clone(),
-                        });
+                    ValueOps::Load | ValueOps::Alloc => {
+                        // NB: Load takes a state edge _and_ returns one. We
+                        // could relax this later (making it easier to reorder
+                        // loads, etc.), but only after confirming that it is
+                        // safe wrt the model we're using. For example, in
+                        // WebAssembly, loads can trap.
+                        let dest_var = self.analysis.intern.intern(dest);
+                        let mut ops = convert_args(args, &mut self.analysis, &mut self.store, pos)?;
+                        ops.push(self.store[&self.analysis.state_var]);
+                        let expr = BasicExpr::Op(*op, ops, op_type.clone());
```

Comment:
Should `Load` be another variant of `BasicExpr`? It might require special handling due to state edges unlike other ops

Reply:
What kind of special handling does would that be? I think most kinds of special handling could happen by pattern matching, what does the extra variant buy?

Reply:
you are right